python/lsst/ts/ess/common/sensor/base_sensor.py

```diff
@@ -69,3 +69,16 @@ async def extract_telemetry(self, line: str) -> TelemetryDataType:
             channels.
         """
         pass
+
+    def __repr__(self) -> str:
+        return self.__str__()
+
+    def __str__(self) -> str:
+        vars_list = [
+            f"{var}={val!r}"
+            for var, val in vars(self).items()
+            if var not in ["log", "terminator"]
+        ]
+        vars_str = ", ".join(vars_list)
+        st = f"{type(self).__name__}<{vars_str}>"
+        return st
```

Comment:
Please consider a common base class or a free function, so you don't have to repeat all of this. A free function would work just fine.

Reply:
This was done in two common base classes: one for the sensors and one for the devices. Do you suggest that I create another common base class above those just for having one definition of `__repr__`?

Reply:
I would not implement a common base class just for this one function. If they don't have any obvious commonality other than `__repr__` then that is artificial and likely to lead to confusion. I'd use a free function in that case.